llvm/lib/Transforms/InstCombine/InstCombineAddSub.cpp

```diff
@@ -1694,6 +1694,26 @@ Instruction *InstCombinerImpl::visitAdd(BinaryOperator &I) {
     return BinaryOperator::CreateOr(LHS, Zext);
   }
 
+  {
+    Value *Cond, *Ext;
+    Constant *C;
+    // (add X, (sext/zext (icmp eq X, C)))
+    //    -> (select (icmp eq X, C), (add C, (sext/zext 1)), X)
```

Comment:
Can you please provide alive2 proof? I am worry about undef-safety.

Reply:
Yeah sorry, updated desc.